cmd/swarmctl/service/common.go

```diff
@@ -13,28 +13,63 @@ func getService(ctx context.Context, c api.ControlClient, input string) (*api.Se
 	rg, err := c.GetService(ctx, &api.GetServiceRequest{ServiceID: input})
 	if err != nil {
 		// If any error (including NotFound), ListServices to match via ID prefix and name prefix.
-		rl, err := c.ListServices(ctx,
-			&api.ListServicesRequest{
-				Filters: &api.ListServicesRequest_Filters{
-					NamePrefixes: []string{input},
-				},
-			},
-		)
+		service, err := getServiceByName(ctx, c, input)
```

Comment:
It's probably better to lookup by prefix, then let the caller know the name is ambiguous.